languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -28,11 +31,25 @@ def clear
       # @param name [String]
       # @raise [PolarFileExtensionError] if provided filename has invalid extension.
       # @raise [PolarFileNotFoundError] if provided filename does not exist.
-      def load_file(name)
+      def load_file(name) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         raise PolarFileExtensionError unless ['.pol', '.polar'].include? File.extname(name)
         raise PolarFileNotFoundError, name unless File.file?(name)
```

Comment:
as another nit, since we aren't lazy loading we should now just try reading the file here and handle errors instead of checking for existence and then reading the file if it exists.